File: app/src/main/java/oop2/ex03.java
package oop2;

public class ex03 {
    interface Movable {
        void move();
    }

    interface Attackable {
        void attack();
    }

    static class Hero implements Movable, Attackable {
        @Override
        public void move() {
            System.out.println("Hero moves fast");
        }

        @Override
        public void attack() {
            System.out.println("Hero attacks");
        }
    }

    static class Monster implements Movable, Attackable {
        @Override
        public void move() {
            System.out.println("Monster moves slowly");
        }

        @Override
        public void attack() {
            System.out.println("Monster attacks");
        }
    }

    public static void main(String[] args) {
        Hero hero = new Hero();
        Monster monster = new Monster();

        hero.move();
        hero.attack();
        monster.move();
        monster.attack();
    }
}

File: app-kotlin/src/main/kotlin/oop2/ex03.kt
package oop2

class ex03 {
    interface Movable {
        fun move()
    }

    interface Attackable {
        fun attack()
    }

    class Hero : Movable, Attackable {
        override fun move() {
            println("Hero moves fast")
        }

        override fun attack() {
            println("Hero attacks")
        }
    }

    class Monster : Movable, Attackable {
        override fun move() {
            println("Monster moves slowly")
        }

        override fun attack() {
            println("Monster attacks")
        }
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val hero = Hero()
            val monster = Monster()
            hero.move()
            hero.attack()
            monster.move()
            monster.attack()
        }
    }
}
